src/pyats/contrib/plugins/topoup_plugin/topoup.py

```diff
@@ -86,9 +87,24 @@ def pre_job(self, task):
             ckwargs = {'start_time': start_time, 'timeout': timeout, 'interval': interval},
             ikwargs = [{'device':self.runtime.testbed.devices[dev]} for dev in self.runtime.testbed.devices])
 
-        if not (pcall_output[0] and pcall_output[1]):
+        # Create Summary
+        log.info(banner("Devices' connection trials summary"))
+
+        failed = False
+
+        for res, dev, count in pcall_output:
+            if res is False:
+                failed = True
+                log.warning("Device '{device}' connectivity check failed after '{count}' trial(s)".format(
+                    device=dev, count=count))
+            else:
+                log.info("Device '{device}' connectivity passed after '{count}' trial(s)".format(
```

Comment:
can we have them grouped

the passed one first 
then the failed one

Reply:
![image](https://user-images.githubusercontent.com/33664528/95100610-493b9200-06ff-11eb-8d29-90ae63928522.png)

![image](https://user-images.githubusercontent.com/33664528/95100538-32953b00-06ff-11eb-9bba-38606f2a9ad9.png)